File: Objective_C/PersonitaObjectiveC/PersonitaObjectiveC/Personita.m
//
//  Personita.m
//  PersonitaObjectiveC
//
//  Created by Elías Jiménez on 21/08/24.
//

#import "Personita.h"

@implementation Personita

- (id) init {
    self = [super init];
    _nombre = @"Sin nombre";
    _edad = 0;
    return self;
}

- (id) initWithNombre: (NSString*) nombre {
    self = [super init];
    _nombre = nombre;
    _edad = 0;
    return self;
}

- (void) saludar {
   NSLog(@"Hola, soy %@ y tengo %d años\n", _nombre, _edad);
}

@end

File: swift/PersonitaEjemplo/PersonitaEjemplo/Personita.swift
//
//  Personita.swift
//  PersonitaEjemplo
//
//  Created by Elías Jiménez on 19/08/24.
//

import Foundation

class Personita {
    let nombre: String
    var edad: Int
    
    
    
    init(nombre: String) {
        self.nombre = nombre
        self.edad = 0
    }
    
    init(nombre: String, edad: Int) {
        self.nombre = nombre
        self.edad = edad
    }
    
    func crecer() {
        edad += 1
    }
    
    func crecer(x: Int) {
        edad += x
    }
    
    func saludar() {
        print("Hola soy \(nombre) y tengo \(edad) años")
    }
    
}
